test/powershell/Modules/Microsoft.PowerShell.Utility/NewTemporaryFile.Tests.ps1

```diff
@@ -15,19 +15,100 @@
 #>
 
 Describe "NewTemporaryFile" -Tags "CI" {
+    $defaultTemporaryFileExtension = '.tmp'
 
     It "creates a new temporary file" {
         $tempFile = New-TemporaryFile
+        try
+        {
+            
+            Test-Path $tempFile | Should be $true
+            $tempFile | Should BeOfType System.IO.FileInfo
+            $tempFile.Extension | Should be $defaultTemporaryFileExtension
+        }
+        finally
+        {
+            Remove-Item $tempFile -ErrorAction SilentlyContinue -Force
+        }
+    }
 
-        Test-Path $tempFile | Should be $true
-        $tempFile | Should BeOfType System.IO.FileInfo
+    It "creates a new temporary file with a specific extension using the -Extension parameter" {
```

Comment:
This test has repeated code, you should use the `-Testcases` feature of Pester instead.